pkg/commands/kustomizationfile.go

```diff
@@ -203,6 +203,9 @@ func isCommentOrBlankLine(line []byte) bool {
 // If the field wasn't in the original kustomization.yaml file or wasn't added,
 // an empty []byte is returned.
 func marshalField(field string, kustomization *types.Kustomization) ([]byte, error) {
+	if field == "apiVersion" {
```

Comment:
Can you add a comment explaining why you need to do this? I don't quite understand.

Reply:
ditto

Reply:
In current `marshalField` function, the filed is assumed to be in a pattern that the correct field name can be get by `strings.Title`, such as `kind` --> `Kind`.  However, `apiVersion` doesn't follow this pattern. Let me know if the assumption is to strong. 

In addition, I found that `KIND` or `Kind` are supported by yaml. So I assume we also need to support them when we parse the kustomization.yaml file. That's why I added a `(?i)` in the regexp.

Reply:
I refactored the code for finding the matching field. See if it looks better.